preon-binding/src/main/java/org/codehaus/preon/codec/IntegerType.java

```diff
@@ -0,0 +1,338 @@
+//========================================================================
+//
+//                       U N C L A S S I F I E D
+//
+//========================================================================
+//  Copyright (c) 2021 Chesapeake Technology International Corp.
+//  ALL RIGHTS RESERVED
+//  This material may be reproduced by or for the U.S. Government
+//  pursuant to the copyright license under the clause at
+//  DFARS 252.227-7013 (OCT 1988).
+//=======================================================================
+
+package org.codehaus.preon.codec;
+
+import org.codehaus.preon.DecodingException;
+import org.codehaus.preon.buffer.BitBuffer;
+import org.codehaus.preon.buffer.ByteOrder;
+import org.codehaus.preon.channel.BitChannel;
+
+import java.io.IOException;
+
+/**
+ * Integer types of numerics, since some things can be done with integer types that you wouldn't want to do with a
+ * floating point type.
+ *
+ * @author Copyright &#169; 2021 Chesapeake Technology International Corp.
+ */
+public enum IntegerType implements IIntegerType
+{
+    Integer
+            {
+                @Override
+                public Object decodeLeb128(BitBuffer buffer) throws DecodingException
+                {
+                    int result = 0;
+                    int cur;
```

Comment:
Is there a reason we really want to do that? I'd need another way of building the result, then, since right now it needs cur to be the same size because it moves the bits over so the bits in cur line up with where they go in the result and uses a bitwise or to put them in.

Reply:
Ah yeah my mistake on sizing.  And yeah could move to long, but no there really isnt a strong reason to do so.